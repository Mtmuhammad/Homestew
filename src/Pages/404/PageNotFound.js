import { React, useEffect } from "react";
import { useNavigate } from "react-router-dom";
import "./PageNotFound.scss";

const PageNotFound = () => {
  const navigate = useNavigate();

  useEffect(() => {
    setTimeout(() => navigate("/"), 3000);
  }, [navigate]);

  return (
    <div id="notfound">
      <div className="notfound">
        <div className="notfound-404">
          <h3>Oops! Page not found</h3>
          <h1>
            <span data-testid="span">4</span>
            <span data-testid="span">0</span>
            <span data-testid="span">4</span>
          </h1>
        </div>
        <h2>we are sorry, but the page you requested was not found</h2>
      </div>
    </div>
  );
};

export default PageNotFound;
