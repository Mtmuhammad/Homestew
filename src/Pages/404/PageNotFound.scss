@use "../../Styles/abstracts/mixins" as *;
@use "../../Styles/abstracts/variables" as *;

#notfound {
  position: relative;
  height: 100vh;
}

.notfound {
  max-width: 520px;
  width: 100%;
  line-height: 1.4;
  text-align: center;
  position: absolute;
  left: 50%;
  top: 45%;
  transform: translate(-50%, -50%);

  &-404 {
    position: relative;
    height: 24rem;

    & h3 {
      position: relative;
      font-size: 1.8rem;
      font-weight: 700;
      text-transform: uppercase;
      color: #262626;
      margin: 0px;
      letter-spacing: 3px;
      padding-left: 20px;
    }

    & h1 {
      position: absolute;
      left: 50%;
      top: 50%;
      transform: translate(-50%, -50%);
      font-size: 30rem;
      font-weight: 900;
      margin-top: 2rem;
      color: $primary-green;
      text-transform: uppercase;
      letter-spacing: -40px;
      margin-left: -20px;

      & span {
        text-shadow: -8px 0px 0px #fff;
      }
    }
  }

  & h2 {
    font-size: 2.5rem;
    font-weight: 400;
    text-transform: uppercase;
    color: #000;
    margin-top: 5rem;
    margin-bottom: 25px;
  }
}
