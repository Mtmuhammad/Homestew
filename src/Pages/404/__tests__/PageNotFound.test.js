import React from "react";
import { render, screen } from "@testing-library/react";
import PageNotFound from "../PageNotFound";
import {BrowserRouter} from "react-router-dom"

describe("smoke and snapshot tests", () => {
  test("should render without crashing", () => {
    render(<PageNotFound />, {wrapper: BrowserRouter});
  });

  test("should match snapshot", () => {
    const { asFragment } = render(<PageNotFound />, {wrapper: BrowserRouter});
    expect(asFragment()).toMatchSnapshot();
  });
});

test("should render h3", () => {
  render(<PageNotFound />, {wrapper: BrowserRouter});
  const heading = screen.getByRole("heading", {
    name: /oops! page not found/i,
  });
  expect(heading).toBeInTheDocument();
});

test("should render h2", () => {
  render(<PageNotFound />, {wrapper: BrowserRouter});
  const heading = screen.getByRole("heading", {
    name: /we are sorry, but the page you requested was not found/i,
  });
  expect(heading).toBeInTheDocument();
});

test("should render spans", () => {
  render(<PageNotFound />, {wrapper: BrowserRouter});
  const spans = screen.getAllByTestId("span");
  expect(spans.length).toBe(3);
});
