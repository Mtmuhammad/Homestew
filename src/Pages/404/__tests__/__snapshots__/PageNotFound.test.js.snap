// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`smoke and snapshot tests should match snapshot 1`] = `
<DocumentFragment>
  <div
    id="notfound"
  >
    <div
      class="notfound"
    >
      <div
        class="notfound-404"
      >
        <h3>
          Oops! Page not found
        </h3>
        <h1>
          <span
            data-testid="span"
          >
            4
          </span>
          <span
            data-testid="span"
          >
            0
          </span>
          <span
            data-testid="span"
          >
            4
          </span>
        </h1>
      </div>
      <h2>
        we are sorry, but the page you requested was not found
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
